send/netbox

```diff
@@ -0,0 +1,250 @@
+		my $user = $perun_users->{$_};
+		my @alphanumeric = ('a'..'z', 'A'..'Z', 0..9);
+		my $randpassword = join '', map $alphanumeric[rand @alphanumeric], 0..10;
+		$user->{"password"} = $randpassword;
+		my @groups_ids = ();
+		for my $name (@{$user->{"groups"}}) {
+			my $group_id = $perun_groups->{$name};
+			push @groups_ids, int($group_id);
+		}
+		$user->{"groups"} = \@groups_ids;
+		call_server($USERS_API, $TYPE_POST, $user);
+	}
+}
+
+# Update existing netbox users
+foreach (keys %$netbox_users_hash) {
+	my @users_to_update = ();
+	if (exists $perun_users->{$_}) {
+		my $user = $perun_users->{$_};
+		$user->{"id"} = $netbox_users_hash->{$_}->{"id"};
+		my @groups_ids = ();
+		for my $name (@{$user->{"groups"}}) {
+			my $group_id = $perun_groups->{$name};
+			push @groups_ids, int($group_id);
+		}
+		$user->{"groups"} = \@groups_ids;
+		push @users_to_update, $user;
+	} else {
+		my $user = $netbox_users_hash->{$_};
+		$user->{"is_active"} = \0;
+		$user->{"groups"} = [];
+		push @users_to_update, $user;
+	}
+	call_server($USERS_API, $TYPE_PATCH, \@users_to_update);
+}
+
+###################### SUBROUTINES ######################
+
+#Name:
+# callServer
+#-----------------------
+#Parameters:
+# url     - url of server and method address to call
+# type    - GET, POST or PATCH
+# content - hash content of request, will be encoded to json output
+#-----------------------
+#Returns:
+# resolved status of call
+# 0 means everything is ok and done
+# die in other case (hard error, server error or other error)
+#-----------------------
+#Description:
+# This method calls specific url with GET, POST or PATCH connection type and predefined content in json.
+# After that it check response, if there is correct answer or error.
+# If there is no server error, it returns resolved status of this call.
+#-----------------------
+sub call_server {
+	my $url = shift;
+	my $type = shift;
+	my $content = shift;
+
+	my $json_content = JSON->new->utf8->encode( $content );
+
+	my $server_response = create_connection( $url, $type, $json_content );
+	return check_server_response( $server_response );
+}
+
+#Name:
+# createConnection
+#-----------------------
+#Parameters:
+# url     - url of server and method address to call
+# type    - GET, POST or PATCH
+# content - hash content of request, will be encoded to json output
+#-----------------------
+#Returns: Response of server on our request.
+#-----------------------
+#Description:
+# This method just takes all parameters and creates connection to the server.
+# Then returns it's repsonse.
+#-----------------------
+sub create_connection {
+	my $url = shift;
+	my $type = shift;
+	my $content = shift;
+
+	my $headers = HTTP::Headers->new;
+	$headers->header('Content-Type' => 'application/json');
+	$headers->header('Accept' => 'application/json,text/json');
+	$headers->header('Authorization' => 'Token ' . $TOKEN);                                                       #ASI DOBRE
```

Comment:
You probably forgot to delete the comment

Reply:
Fiked